### PR TITLE
Ravi/dashboard 1

### PR DESCRIPTION
**Pull request by raviinder**
Code to add 
1. user Dashboard code
2. Add User Saved Searches. 
3. Delete User Saved Searches. 
4. Add User Saved Records. 
5. Delete User Saved Records.

### CODE REVIEW

**Comment by johnweng001** [comment]
Here is some comments:
1.  Line 16 in dashboard.ts, switch(+dc),  + is not necessary for enum
2. Duplicate api calls is observed when dashboard component is first time rendered. The reason is both userID and language changes triggered. As a workaround, the language can be moved outside the component an pass in thru props (from dashboard-panel). Or we can use locale context (but I don't see context used in general app). So we don't need two useEffect hooks.
3. In dashboard component, the function inside useEffect (both getAnnouncement and getCommunityAndSavedRecords) has dependency in both userID and language. In that sense, the function itself should be in the useEffect dependency list as well.
That means, the function needs to be wrapped in useCallback. (this is recommended by eslint rule: react-hooks/exhaustive-deps)

File: src/common/dashboard.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export enum DASHBOARD_CALLS {
+    ANNOUNCEMENT,
+    COMMUNITY,
+    SAVED_RECORDS
+}
+
+const getAPIURL = (dc: DASHBOARD_CALLS, endpointUrl: string) => {
+    let url = "";
+    switch (+dc) {
```
Comment (johnweng001) [inline src/common/dashboard.ts]: + is not necessary

File: src/components/dashboard/dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+    useEffect(() => {
+        if (announcementLoading) {
+            getAnnouncement();
+        }
+    }, [announcementLoading, userID]);
+
+    useEffect(() => {
+        getAnnouncement();
+    }, [language]);
```
Comment (johnweng001) [inline src/components/dashboard/dashboard.tsx]: when the component first time loaded, both language and userID changes will trigger the getAnnouncement called twice. (duplicate). language can be moved outside the component and passed from props, as a workaround.

File: src/components/dashboard/dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+const Dashboard = () => {
+    const { t } = useTranslation();
+    const language = t('app.language');
+    const [userID, setUserID] = useState("cf9bfa6f-5837-42f9-9eeb-c15035c3c752");
+
+    //#region accouncements
+    const [announcementLoading, setAnnouncementLoading] = useState(true);
+    const [announcement, setAnnouncement] = useState("");
+
+    const getAnnouncement = () => {
```
Comment (johnweng001) [inline src/components/dashboard/dashboard.tsx:60]: getAnnoucement has dependency in both language and userID. So it needs to be defined in useEffect event list as well.
And the function itself can be wrapped in useCallBack ( see recommendation from eslint rule: react-hooks/exhaustive-deps)

File: src/components/dashboard/dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    }, [savedRecordsLoading]);
+
+    useEffect(() => {
+        getCommunityAndSavedRecords(DASHBOARD_CALLS.SAVED_RECORDS, '', setsavedRecordsLoading, setSavedRecords);
+    }, [language]);
+
+    //#endregion
+
+
+    const getCommunityAndSavedRecords = (dc: DASHBOARD_CALLS, url, loading, setData) => {
```
Comment (johnweng001) [inline src/components/dashboard/dashboard.tsx]: move the function definition before the useEffect where is referred.

File: src/common/dashboard.ts (hunk 1 above)
Comment (raviinder) [reply src/common/dashboard.ts]: Accepted.

**Comment by jvanulde** [comment]
We should list the issues that this PR addresses. You can select them in the right column in the Development panel.

**Pull request closed without merging**